entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -543,9 +555,78 @@ public void total(
     responseObserver.onCompleted();
   }
 
+  @Override
+  public void deleteEntities(
+      DeleteEntitiesRequest request, StreamObserver<DeleteEntitiesResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    if (validateDeleteEntitiesRequest(request, requestContext)) {
+      responseObserver.onError(new ServiceException("Invalid request provided"));
+      return;
+    }
+
+    List<String> entityIds = getEntityIdsToDelete(requestContext, request);
+    if (entityIds.size() == 0) {
+      LOG.info("{}. No entities found to delete", request);
+      responseObserver.onNext(DeleteEntitiesResponse.newBuilder().build());
+      responseObserver.onCompleted();
+      return;
+    }
+
+    if (entityIds.size() > maxEntitiesToDelete) {
+      LOG.info(
+          "{}. Number of ids to delete exceeds the maximum limit: Entity Ids limit exceeded",
+          request);
+      responseObserver.onError(new RuntimeException("Entity Ids limit exceeded"));
+      return;
+    }
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Deleting entity of type: {} and ids: {}", request.getEntityType(), entityIds);
+    }
+
+    Optional<String> tenantId = requestContext.getTenantId();
+    this.entitiesCollection.delete(
+        DocStoreConverter.transform(tenantId.get(), request.getEntityType(), entityIds));
+    responseObserver.onNext(DeleteEntitiesResponse.newBuilder().addAllEntityIds(entityIds).build());
+    responseObserver.onCompleted();
+  }
+
+  private List<String> getEntityIdsToDelete(
+      RequestContext requestContext, DeleteEntitiesRequest request) {
+    List<Expression> selections =
+        List.of(
+            Expression.newBuilder()
+                .setColumnIdentifier(
+                    ColumnIdentifier.newBuilder().setColumnName(ENTITY_ID_ATTRIBUTE_ID).build())
+                .build());
+    EntityQueryRequest entityQueryRequest =
+        EntityQueryRequest.newBuilder()
+            .setEntityType(request.getEntityType())
+            .setFilter(request.getFilter())
+            .build();
+    Query query = entityQueryConverter.convertToEDSQuery(requestContext, entityQueryRequest);
```

Comment:
I'm guessing this logic is duped from the query API. Once we map the delete to a query request, should be able to share the query and response processing logic.

Reply:
It's not precisely duped. It's taking some bits but not all. The problem is query Api requires selection as a list of Id attributes defined in the attributes map. Being generic we don't have that while deleting entities. So i am skipping the attribute mapping phase of result creation. Let me know if there is any other way?

Reply:
> The problem is query Api requires selection as a list of Id attributes defined in the attributes map

Not sure I'm following - you need the attributes to filter anyway, right? So I would assume you'd need to pull these from config. If you can get away without having to configure them, then great, but it just means a different unit would be shareable (the part up until result mapping).

Reply:
EDS coupling is anyways there in execute api of EQS as well as total api. The difference delete api had is in the response translation. EQS other apis translate responses based on attributes provided, but we don't need that here. So that's why the difference. But I am sensing based on your comment that, this distinction was also mixed sometime before. let me try the without this mapping

Reply:
> Couldn't you just send a query to EDS by translating the filters and then getting the response entities and taking the IDs out of that? The usage of the constant is only necessary because you're directly calling doc store it looks like?

EQS doesn't call EDS directly. I don't think we should add that. We can only use EQS `execute` api only but my concern is around how to add the correct selection filter for entityId. Because those `execute` assumed that they will from the attribute map only. Can I assume the entity id attribute will always be `{ENTITY_TYPE}.id`. I don't see anything that works on this.

Reply:
> EQS doesn't call EDS directly. I don't think we should add that.

It can and really should - we're saying we'd prefer to couple their implementations (e.g. sharing constants and duplicating logic and have two different services reading and writing the same data) rather than than having one data owner and another that presents a different API to the same data? That's a majorly problematic design. I'm not saying we need to fix that everywhere, we've already dug ourselves in deep, but if we're avoiding that as an easy solution to the problem we need not be.

> Can I assume the entity id attribute will always be {ENTITY_TYPE}.id.

No - if we don't want to rely on EDS (see above, we should be!), we'll need to maintain a map similar to gateway service https://github.com/hypertrace/gateway-service/blob/main/gateway-service/src/main/resources/configs/common/application.conf#L53

Reply:
Why aren't we using the other `queryConverter.convert(request, requestContext);` we have in the query implementation?

https://github.com/hypertrace/entity-service/blob/c1ab8baf0077c92d8c5aa05e92507f55b3620f31/entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java#L157

We are planning to remove the entire `convertToEDSQuery` method, since it was deprecated (since aggregation support didn't exist in Postgres before)

Reply:
Yes. Ronak is working on implementing the new Query APIs for Postgres and is expected to complete by this week. Most of the functionality is already there. And, the functionality needed for this purpose is already supported.
You may have to cut a latest release of document-store and use it here.

If everything goes well, we will have the `queryAggregationEnabled` flag removed soon, defaulting it to `true`. So, instead of surrounding this block I'd suggest that we use the new (document-store) query API (that directly converts the EQS query to document store query without the intermediate EDS query conversion). Better to get an approval from @kotharironak 

You could do some refactoring to extract out the common code block between `execute` and this method.